requirements.txt

```diff
@@ -7,10 +7,9 @@ UnittestZero
 certifi==0.0.8
 chardet==2.1.1
 execnet==1.1
-py==1.4.15
-pytest==2.3.5
+py==1.4.26
+pytest==2.7.0
 pytest-mozwebqa
-pytest-xdist==1.8
```

Comment:
```IRC
19:38 davehunt: jgmize: we install it in Jenkins via an infrastructure requirements script
19:38 davehunt: it's not a requirement of the test repository though
19:39 jgmize: davehunt: where is the infrastructure requirements script?
19:39 jgmize: (other thank jenkins itself) -- I mean is there a github repo?
19:39 davehunt: jgmize: https://webqa-ci.mozilla.com/configfiles/editConfig?id=org.jenkinsci.plugins.managedscripts.ScriptConfig1422993554111
19:39 davehunt: admin privs required, as it's global config
19:39 jgmize: I can't access that script
19:40 davehunt: #!/bin/bash -ex
19:40 davehunt: pip install pytest-rerunfailures==0.03
19:40 davehunt: pip install pytest-timeout==0.3
19:40 davehunt: pip install pytest-xdist==1.8
19:40 davehunt: contents is ^
19:40 jgmize: why isn't that public?
19:40 davehunt: because it's global config
19:40 davehunt: we can't make it public without making all global config public
19:40 jgmize: why put it there at all then?
19:40 davehunt: the permissions are not fine grained enough
19:41 davehunt: jgmize: because we need to centrally maintain global infrastructure requirements
19:41 jgmize: why?
19:41 davehunt: for our sanity :)
19:41 davehunt: all jobs in jenkins call that script
19:42 davehunt: we have dozens of jobs in Jenkins, adding that to each one would be a pain to maintain
19:42 davehunt: previously we had these stored in a repository, but that was overkill
19:43 davehunt: jgmize: you shouldn't have any failures without pytest-xdist, your tests would just run in serial
19:43 davehunt: that's the default for pytest
19:44 jgmize: but you're locking every set of tests into that set of requirements, and making it difficult to replicate outside of that centrally managed environment. My personal preference, which you can obviously feel free to disregard, would be to maintain requirements for the job in the repo for the job itself. Running the tasks in serial is decidedly non-optimal,
19:44 jgmize: extending an already long development cycle needlessly
19:44 davehunt: jgmize: you're running them locally?
19:45 jgmize: I'm running them on ci.masterfirefoxos.com because I don't have access to create jobs to run against demo environments anywhere else.
19:45 davehunt: I wasn't aware we had another Jenkins instance running these tests
19:46 davehunt: jgmize: we have a staging instance: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/
19:46 davehunt: you can have full admin there
19:46 jgmize: davehunt: thanks, I would appreciate that.
19:46 davehunt: let me know your username and I'll grant admin
19:49 davehunt: jgmize: I don't have a strong objection to the test repository depending on pytest-xdist, just that it's technically neither a requirement or a dependency. If we add it and pin it, it's another requirement to maintain. I don't think it's updated frequently though.
19:52 jgmize: davehunt: username is jgmize, thanks
19:52 davehunt: jgmize: already done :)
```